src/System.Text.Json/tests/Serialization/CustomConverterTests.Dictionary.cs

```diff
@@ -86,5 +87,119 @@ public static void CustomDictionaryConverter()
                 Assert.Equal(json, jsonSerialized);
             }
         }
+
+        [Fact]
+        public static void CustomDictionaryConverterContravariant()
+        {
+            const string json = @"{""Key1"":1,""Key2"":2}";
+
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new IDictionaryConverter(10));
+
+            Dictionary<string, long> dictionary = JsonSerializer.Deserialize<Dictionary<string, long>>(json, options);
+            Assert.Equal(11, dictionary["Key1"]);
+            Assert.Equal(12, dictionary["Key2"]);
+
+            string jsonSerialized = JsonSerializer.Serialize(dictionary, options);
+            Assert.Equal(json, jsonSerialized);
+        }
+
+        [Fact]
+        public static void ClassWithCustomDictionaryConverterTest()
+        {
+            const string json = @"{""MyInt"":32,""MyDictionary"":{""Key1"":1,""Key2"":2},""MyString"":""Hello""}";
+
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new IDictionaryConverter(10));
+
+            ClassWithCustomDictionaryConverter dictionary = JsonSerializer.Deserialize<ClassWithCustomDictionaryConverter>(json, options);
+            Assert.Equal(11, dictionary.MyDictionary["Key1"]);
+            Assert.Equal(12, dictionary.MyDictionary["Key2"]);
+            Assert.Equal(32, dictionary.MyInt);
+            Assert.Equal("Hello", dictionary.MyString);
+
+            string jsonSerialized = JsonSerializer.Serialize(dictionary, options);
+            Assert.Equal(json, jsonSerialized);
+        }
+
+        private class ClassWithCustomDictionaryConverter
+        {
+            public int MyInt { get; set; }
+            public Dictionary<string, long> MyDictionary { get; set; }
+            public string MyString { get; set; }
+        }
+
+        private class IDictionaryConverter : JsonConverter<IDictionary<string, long>>
+        {
+            private long _offset;
+
+            public IDictionaryConverter(long offset)
+            {
+                _offset = offset;
+            }
+
+            public override bool CanConvert(Type typeToConvert)
```

Comment:
Can this method just be `return typeToConvert == typeof(Dictionary<string, long>)`?

Reply:
yes it can